### PR TITLE
Use insight short ID in `insightLogic` key also on the insight page

### LINKED ISSUE #8478: Bug: Wrong insight type loading when editing from dashboard
## Bug description

Spotted this for the first time on a call with @andyvan-ph, but he says it's happened a few times to him before so it's definitely replicable. 

If you start from a dashboard with multiple types of insight and click 'Edit' on one type (e.g. Trends), then click Back to return to the dashboard and click 'Edit' on another type (e.g. Paths) then PostHog attempts to load a Trends insight again. I _think_ it's trying and failing to load it with the same information, because the insight will never successfully load if left -- but if you click to the correct type of insight, suddenly it loads instantly. 

This is a little hard to explain, [so here's a video.](https://drive.google.com/file/d/18xpW3VMQoNtXb_66eK61fUA3L7HcLtvl/view?usp=sharing)

## Expected behavior

I'd expect it to load the correct insight. 

## How to reproduce

1. Open a dashboard. Here's what I used. 
2. Click 'Edit' on the Pageviews trends insight. Then click back to return to the dashboard. 
3. Click 'Edit' on the Paths insight. 

## Environment

- [x] PostHog Cloud
- [ ] self-hosted PostHog (ClickHouse-based), version/commit: _please provide_
- [ ] self-hosted PostHog (Postgres-based, legacy), version/commit: _please provide_

## Additional context

Taggin @andyvan-ph who can provide more examples and context, if needed.

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

Fixes #8478. Currently if `insightLogic`'s prop `syncWithUrl` is true, the logic's key is `scene` – otherwise it's `dashboardItemId` (actually the relevant insights "short ID"). So the insights page uses the same logic for all insights, while dashboard items are separated. However, this makes it easier for the insight page to confuse two insights – one race condition and it fails like in the mentioned issue. This adds the insight ID to the `syncWithUrl` case's key too to ensure separation.
I was actually thinking ideally that insight ID would be the _only_ key, but when I tried that out, I encountered some bugs.
BTW not sure if we really need to literally `syncWithUrl` anymore after your recent changes @mariusandra?

## How did you test this code?

Reproduced with steps provided by Joe, but didn't automate.

### CODE REVIEW

File: frontend/src/scenes/insights/sharedUtils.ts [hunk 1]
```diff
@@ -9,14 +9,18 @@ import { FilterType, InsightLogicProps, InsightType } from '~/types'
...
+    return (props) => {
         if (!('dashboardItemId' in props)) {
             throw new Error('Must init with dashboardItemId, even if undefined')
         }
-        return props.syncWithUrl ? sceneKey : props.dashboardItemId || defaultKey
+        let key = props.dashboardItemId || defaultKey
+        if (props.syncWithUrl) {
+            key = `${sceneKey}-${key}`
+        }
+        return key
```
Comment (mariusandra) [inline frontend/src/scenes/insights/sharedUtils.ts:21]: We don't use a custom `sceneKey` anywhere. Mights as well get rid of the parameter.

Some part of me really wants to nit to use a `? :` here to save precious lines and whitespace.

**Review by mariusandra — APPROVED** [review]
I think this makes sense, though we can clean up a little bit as well. I ran this locally and couldn't find a way to make this crash.

The `syncWithUrl` prop is needed, as `insightLogic` is still the one that does the syncing. All my recent url changes were around actually getting the syncing into this logic, and out of all the others. We could rename it `isScene`, but 🤷.

File: frontend/src/scenes/insights/sharedUtils.ts (hunk 1 above)
Comment (Twixes) [reply frontend/src/scenes/insights/sharedUtils.ts]: Makes sense. Also, switched to that sweet ternary operator. 🍍

Comment (mariusandra) [reply frontend/src/scenes/insights/sharedUtils.ts]: such neat

**Pull request merged**